### PR TITLE
Removed HTML tags from Markdown trait export

### PR DESCRIPTION
**Pull request by Mailaender**
This is a workaround for https://github.com/mkdocs/mkdocs/issues/2400, but I thought it might actually be nicer as Markdown is easier for humans to read compared to hard-coded HTML tags.

### CODE REVIEW

**Comment by pchote** [comment]
Looks reasonable, but have you tested it works? I usually find it takes 5+ attempts to get tables to actually render whenever I try to make one.

The other doc generating commands should be changed to match.

**Comment by Mailaender** [comment]
I tested with @retext-project and @mkDocs locally.

**Review by abcdefg30 — DISMISSED** [review]
Works nicely.

> The other doc generating commands should be changed to match.

That is still open, isn't it?

**Comment by Mailaender** [comment]
If you insist. As MarkDown forbidds linebreaks in tables I had to come up with something else for the Lua page.

**Comment by pchote** [comment]
What do you mean? Do you have a before and after image? Some quick searching suggests that `<br>` should work fine in tables...

**Comment by Mailaender** [comment]
| Test  
line break | Test |
| ----------- |  ----- |

| Test without line break | Test |
| ------------------------------ |  ---- |

**Comment by Mailaender** [comment]
I came up with forced tables for the Lua API to preserve your original layout although I think headerless tables were a really bad choice here. It really gave me a hard time here and it isn't very readable in .md viewers.

**Comment by pchote** [comment]
| Test<br />line break<br />using `<br />` | Test |
| ----------- |  ----- |

**Review by abcdefg30 — COMMENTED** [review]
Github renders the triple \* directly and not as bold text:
grafik

Using only `**text**` fixed the issue for me.

**Comment by Mailaender** [comment]
Should be fixed now.

**Comment by abcdefg30** [comment]
Unfortunately, there seems to be some kind of issue with line breaks(?):
grafik

**Comment by abcdefg30** [comment]
I put the output file I got at https://gist.github.com/abcdefg30/6610bdfaa06e4c00dd209a71dd83aa0c, then you can hopefully easily see the spots with that issue. (Or is that a bug in the GitHub Markdown renderer?)

**Comment by Mailaender** [comment]
A Markdown table is required to have a header, which is why @pchote original design is causing trouble here:

|---|---|
| ***int BuildTime(string type, string queue = nil)*** | Returns the build time (in ticks) of the requested unit type.
An optional second value can be used to exactly specify the producing queue type. |
| ***int Cost(string type)*** |  |
| ***Actor Create(string type, bool addToWorld, LuaTable initTable)*** | Create a new actor. initTable specifies a list of key-value pairs that defines the initial parameters for the actor's traits. |
| ***int CruiseAltitude(string type)*** | Returns the cruise altitude of the requested unit type (zero if it is ground-based). |

won't work.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 1]
```diff
@@ -105,28 +111,28 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 					var hasRequires = required.Any();
 					var isActivity = mi.HasAttribute<ScriptActorPropertyActivityAttribute>();
 
-					Console.WriteLine("<tr><td width=\"50%\" align=\"right\"><strong>{0}</strong>", mi.LuaDocString());
+					Console.Write($"| ***{mi.LuaDocString()}***");
 
 					if (isActivity)
-						Console.WriteLine("<br /><em>Queued Activity</em>");
+						Console.WriteLine("<br />*Queued Activity*");
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: Would changing this help perhaps?
```suggestion
						Console.Write("<br />*Queued Activity*");
```

Comment (abcdefg30) [reply OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: (Because it looks like this is adding another new line, which could be an issue?)

Comment (Mailaender) [reply OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: Fixed it.

**Comment by Mailaender** [comment]
I simply added headers with contents as GitHub flavored Markdown and many others don't support headerless tables. https://stackoverflow.com/questions/17536216/create-a-table-without-a-header-in-markdown

**Comment by pchote** [comment]
GitHub flavored markdown *does* support headerless table, I demonstrated this in the comment above: https://github.com/OpenRA/OpenRA/pull/19403#issuecomment-843599548.

Edit: I see, that *was* the header.

**Comment by matjaeck** [comment]
`<code>highlighted within a table!</code>`

**Comment by matjaeck** [comment]
Since the previous comment has been marked off-topic: 

An alternative way to highlight code in HTML-tables is:
``` html
<table>
<tr><th>Property</th><th>Default Value</th><th>Type</th><th>Description</th></tr>
<tr><td>ValidTypes</td><td></td><td>Set of String</td><td>Accepted <code>DeliversCash</code> types. Leave empty to accept all types. </td></tr>
<tr><td>ValidRelationships</td><td>Ally</td><td>PlayerRelationship</td><td>Player relationships the owner of the delivering actor needs. </td></tr>
<tr><td>Sounds</td><td></td><td>Collection of String</td><td>Play a randomly selected sound from this list when accepting cash. </td></tr>
</table>

```

```html
<code>DeliversCash</code>
```

This is valid <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/code">HTML</a>, perhaps someone misread the comment…

Edit: Deleted some formatting changes and actual off-topic moderation-related content since i just wanted to drop this information here and not engage in discussions. Maybe it helps, maybe not.

**Comment by Mailaender** [comment]
But then there is nothing Markdown left and both @GitHub and @Readthedocs take that as input. The HTML is compiled from it.

**Comment by pchote** [comment]
Why do we care about eliminating html from this? Surely the goal should be in having the best quality rendered output.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 2]
```diff
@@ -70,17 +70,20 @@ void IUtilityCommand.Run(Utility utility, string[] args)
...
 
+				Console.WriteLine();
 				Console.WriteLine("### " + name);
-				Console.WriteLine("<table>");
+				Console.WriteLine();
+				Console.WriteLine("| Function | Description |");
+				Console.WriteLine("|----------|-------------|");
 				foreach (var m in members.OrderBy(m => m.Name))
 				{
 					var desc = m.HasAttribute<DescAttribute>() ? m.GetCustomAttributes<DescAttribute>(true).First().Lines.JoinWith("\n") : "";
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: This also needs to change
```suggestion
					var desc = m.HasAttribute<DescAttribute>() ? m.GetCustomAttributes<DescAttribute>(true).First().Lines.JoinWith("<br />") : "";
```
There are two other instances of that in this file, after fixing those as well it renders correctly for me.

**Review by abcdefg30 — COMMENTED** [review]
Lgtm once that is fixed, I don't think we need or should go back now.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs (hunk 2 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: Replaced all instances.

**Review by abcdefg30 — DISMISSED** [review]

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 3]
```diff
@@ -70,17 +70,20 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 				var name = t.GetCustomAttributes<ScriptGlobalAttribute>(true).First().Name;
 				var members = ScriptMemberWrapper.WrappableMembers(t);
 
+				Console.WriteLine();
 				Console.WriteLine("### " + name);
-				Console.WriteLine("<table>");
+				Console.WriteLine();
+				Console.WriteLine("| Function | Description |");
+				Console.WriteLine("|----------|-------------|");
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: We can preserve the right aligned column by using `Console.WriteLine("|---------:|-------------|");`. Likewise for the other table definitions below.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 4]
```diff
@@ -70,17 +70,20 @@ void IUtilityCommand.Run(Utility utility, string[] args)
...
 				foreach (var m in members.OrderBy(m => m.Name))
 				{
-					var desc = m.HasAttribute<DescAttribute>() ? m.GetCustomAttributes<DescAttribute>(true).First().Lines.JoinWith("\n") : "";
-					Console.WriteLine($"<tr><td align=\"right\" width=\"50%\"><strong>{m.LuaDocString()}</strong></td><td>{desc}</td></tr>");
+					var desc = m.HasAttribute<DescAttribute>() ? m.GetCustomAttributes<DescAttribute>(true).First().Lines.JoinWith("<br />") : "";
+					Console.WriteLine($"| ***{m.LuaDocString()}*** | {desc} |");
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: `***` means italic + bold. I think you mean `**` for just bold? Likewise for the other table definitions below.

**Review by pchote — COMMENTED** [review]
Looks fine aside from two points:

**Comment by Mailaender** [comment]
Updated.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 5]
```diff
@@ -105,28 +111,28 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 					var hasRequires = required.Any();
 					var isActivity = mi.HasAttribute<ScriptActorPropertyActivityAttribute>();
 
-					Console.WriteLine("<tr><td width=\"50%\" align=\"right\"><strong>{0}</strong>", mi.LuaDocString());
+					Console.Write($"| ***{mi.LuaDocString()}***");
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: You missed this one!

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 6]
```diff
@@ -105,28 +111,28 @@ void IUtilityCommand.Run(Utility utility, string[] args)
...
-						Console.WriteLine(mi.GetCustomAttributes<DescAttribute>(false).First().Lines.JoinWith("\n"));
+						Console.Write(mi.GetCustomAttributes<DescAttribute>(false).First().Lines.JoinWith("<br />"));
 
 					if (hasDesc && hasRequires)
-						Console.WriteLine("<br />");
+						Console.Write("<br />");
 
 					if (hasRequires)
-						Console.WriteLine($"<b>Requires {(required.Length == 1 ? "Trait" : "Traits")}:</b> {required.JoinWith(", ")}");
+						Console.Write($"***Requires {(required.Length == 1 ? "Trait" : "Traits")}:*** {required.JoinWith(", ")}");
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: And this one.

File: OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs [hunk 7]
```diff
@@ -151,26 +160,26 @@ void IUtilityCommand.Run(Utility utility, string[] args)
...
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/ExtractLuaDocsCommand.cs]: And this one.

**Comment by Mailaender** [comment]
Updated.

**Comment by pchote** [comment]
I'm not sure if the last push broke it or if I just didn't notice it before, but the output of the lua player properties/commands all end up on one line (the other two lua types are output correctly).

**Comment by Mailaender** [comment]
Should be fixed now.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1ba8ef8295057e785907ca0a964dd1aea78698d8

**Pull request merged**